File: Arrays & Hashing/3075. Maximize Happiness of Selected Children/3075.cpp
#pragma GCC optimize("Ofast","Os","unroll-loops","lto","fast-math","march=native","omit-frame-pointer","inline-functions")
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        sort(happiness.begin(),happiness.end());
        long long res=0;
        long long rounds=0;
        int i=happiness.size()-1;

        while(k>0 && i>=0){
            res+=happiness[i]-rounds>=0 ? happiness[i]-rounds : 0;
            rounds++;
            i--;
            k--;
        }
        return res;
    }
};

File: Arrays & Hashing/3075. Maximize Happiness of Selected Children/3075.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        s=res=0
        i=len(happiness)-1
        happiness.sort()
        while k>0 and i>=0:
            res+=happiness[i]-s if happiness[i]-s>=0 else 0
            s+=1
            k-=1
            i-=1
        return res